feature_engine/transformation/date_transformer.py

```diff
@@ -0,0 +1,49 @@
+from typing import List, Optional, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.variable_manipulation import _check_input_parameter_variables
+
+
+class date_feature_extractor(BaseNumericalTransformer):
+    def __init__(
+        self,
+        features: Union[None, int, str, List[Union[str, int]]] = None,
+        variables: Union[None, int, str, List[Union[str, int]]] = None,
+    ) -> None:
```

Comment:
could we add the following parameters?

- `keep_original = False`, which indicates if user wants to drop datetime variables after the transformation or not.
- `missing_values = 'ignore'`, which indicates if missing values should be ignored or alternatively, use 'raise' to throw an error when dataframe has Na